src/l500/l500-device.cpp

```diff
@@ -237,28 +310,85 @@ namespace librealsense
             []() { return std::make_shared<rotation_transform>(RS2_FORMAT_Y8, RS2_STREAM_INFRARED, RS2_EXTENSION_VIDEO_FRAME); }
         );
 
-        depth_ep->register_processing_block(processing_block_factory::create_id_pbf(RS2_FORMAT_Y8, RS2_STREAM_INFRARED));
+        depth_sensor.register_processing_block(processing_block_factory::create_id_pbf(RS2_FORMAT_Y8, RS2_STREAM_INFRARED));
 
-        depth_ep->register_processing_block(
+        depth_sensor.register_processing_block(
             { {RS2_FORMAT_RAW8} },
             { {RS2_FORMAT_RAW8, RS2_STREAM_CONFIDENCE, 0, 0, 0, 0, &l500_confidence_resolution} },
             []() { return std::make_shared<confidence_rotation_transform>(); }
         );
 
-        depth_ep->register_processing_block(processing_block_factory::create_id_pbf(RS2_FORMAT_RAW8, RS2_STREAM_CONFIDENCE));
+        depth_sensor.register_processing_block(processing_block_factory::create_id_pbf(RS2_FORMAT_RAW8, RS2_STREAM_CONFIDENCE));
 
-        return depth_ep;
+        std::shared_ptr< freefall_option > freefall_opt;
+        if( _fw_version >= firmware_version( "1.3.5.0" ) )
+        {
+            depth_sensor.register_option(
+                RS2_OPTION_FREEFALL_DETECTION_ENABLED,
+                freefall_opt = std::make_shared< freefall_option >( *_hw_monitor )
+            );
+        }
+        else
+        {
+            LOG_DEBUG( "Skipping Freefall control: requires FW 1.3.5" );
+        }
+        if( _fw_version >= firmware_version( "1.3.12.9" ) )
+        {
+            depth_sensor.register_option(
+                RS2_OPTION_INTER_CAM_SYNC_MODE,
+                std::make_shared< hw_sync_option >( *_hw_monitor, freefall_opt )
+            );
+        }
+        else
+        {
+            LOG_DEBUG( "Skipping HW Sync control: requires FW 1.3.12.9" );
+        }
+    }
+
+    void l500_device::notify_of_calibration_change( rs2_calibration_status status )
+    {
+        std::time_t now = std::time( nullptr );
+        auto ptm = localtime( &now );
+        char buf[256];
+        strftime( buf, sizeof( buf ), "%T", ptm );
+        AC_LOG( DEBUG, ".,_,.-'``'-.,_,.-'``'-   " << buf << "   status= " << status );
+        for( auto&& cb : _calibration_change_callbacks )
+            cb->on_calibration_change( status );
+    }
+
+    void l500_device::trigger_device_calibration( rs2_calibration_type type )
+    {
+        if( type != RS2_CALIBRATION_DEPTH_TO_RGB )
+            throw not_implemented_exception(
+                to_string() << "unsupported calibration type (" << type << ")" );
+
+        if( !_autocal )
+            throw not_implemented_exception(
+                to_string() << "your firmware version (" << _fw_version
+                            << ") does not support depth-to-rgb calibration" );
+
+        if( _autocal->is_active() )
+            throw wrong_api_call_sequence_exception( "calibration is already active" );
```

Comment:
Maybe: Camera Accuracy Health routine is already active

Reply:
Removed "routine"; otherwise changed.